src/agent.ts

```diff
@@ -726,19 +742,13 @@ export class AgentManager {
     const toolCallId = modelEvent.toolCallId;
     const toolName = modelEvent.toolName;
     const toolInput = modelEvent.input;
-    let parsedToolInput;
-    try {
-      parsedToolInput = JSON.parse(toolInput);
-    } catch (error) {
-      parsedToolInput = {};
-    }
 
     this._agentEvent.emit({
       type: 'tool_call_start',
       data: {
         callId: toolCallId,
         toolName,
-        input: parsedToolInput
+        input: this._formatToolInput(toolInput)
```

Comment:
Thanks @jtpio.

For the record, it seems that in this case the `toolInput` type is any, so calling the `_formatToolInput` could raise an error.
But we can probably try like this and see if it fails sometime.

Reply:
There are a couple of leftover `any` in the code base from the early iteration, but we should indeed get rid of them.

Will take a look to clean that up.